README.md

```diff
@@ -1304,113 +1406,1033 @@ public:
+    map.put("Mali", 28);
+    map.put("Ahmet", 30);
+    map.put("Furkan", 23);
+    query::GreaterLessPredicate<int> greaterThan27Predicate(query::QueryConstants::getValueAttributeName(), 27, false, false);
+    std::vector<std::string> olderThan27 = map.keySet(greaterThan27Predicate);
+    std::cout << "First person:" << olderThan27[0] << ", second person:" << olderThan27[1] << std::endl;
+```
+
+In this example, the code creates a list with the values greater than or equal to "27".
+
+#### 7.7.1.4. Filtering with Paging Predicates
+
+The C++ client provides paging for defined predicates. With its `PagingPredicate` object, you can get a list of keys, values, or entries page by page by filtering them with predicates and giving the size of the pages. Also, you can sort the entries by specifying comparators.
+
+```C++
+    hazelcast::client::IMap<std::string, int> map = hz.getMap<std::string, int>("personMap;");
+    query::PagingPredicate<std::string, int> pagingPredicate(std::auto_ptr<query::Predicate>(new query::GreaterLessPredicate<int>(query::QueryConstants::getValueAttributeName(), 18, false, false)), 5);
+    // Set page to retrieve third page
+    pagingPredicate.setPage(3);
+    std::vector<int> values = map.values(pagingPredicate);
+    //...
+
+    // Set up next page
+    pagingPredicate.nextPage();
+    
+    // Retrieve next page    
+    values = map.values(pagingPredicate);
+    
+    //...
+```
+
+If you want to sort the result before paging, you need to specify a comparator object that implements the `query::EntryComparator` interface. Also, this comparator object should be one of `IdentifiedDataSerializable` or `Portable` or `Custom serializable`. After implementing After implementing this object in C++, you need to implement the Java equivalent of it and its factory. The Java equivalent of the comparator should implement `java.util.Comparator`. Note that the `compare` function of `Comparator` on the Java side is the equivalent of the `sort` function of `Comparator` on the C++ side. When you implement the `Comparator` and its factory, you can add them to the `CLASSPATH` of the server side.  See the [Adding User Library to CLASSPATH section](#adding-user-library-to-classpath). 
+
+Also, you can access a specific page more easily with the help of the `setPage` function. This way, if you make a query for the 100th page, for example, it will get all 100 pages at once instead of reaching the 100th page one by one using the `nextPage` function.
+
+## 7.8. Raw Pointer API
+
+When using C++ client you can have the ownership of raw pointers for the objects you create and return. This allows you to keep the objects in your library/application without any need for copy.
+
+For each container you can use the adapter classes, whose names start with `RawPointer`, to access the raw pointers of the created objects. These adapter classes are found in `hazelcast::client::adaptor` namespace and listed below:
```

Comment:
For each container, you can use the adapter classes, whose names start with `RawPointer`, to access the raw pointers of the created objects. These adapter classes are found in the `hazelcast::client::adaptor` namespace and listed below: